tests/unit/browser/webkit/test_history.py

```diff
@@ -237,24 +218,17 @@ def test_add_item(qtbot, hist):
     list(hist.async_read())
     url = 'http://www.example.com/'
 
-    with qtbot.waitSignals([hist.add_completion_item, hist.item_added],
-                           order='strict'):
-        hist.add_url(QUrl(url), atime=12345, title="the title")
+    hist.add_url(QUrl(url), atime=12345, title="the title")
 
-    entry = history.Entry(url=QUrl(url), redirect=False, atime=12345,
-                          title="the title")
-    assert hist.history_dict[url] == entry
+    assert hist[url] == (url, 'the title', 12345, False)
 
 
 def test_add_item_redirect(qtbot, hist):
```

Comment:
Now that they're more similar, you could combine `test_add_item` and this into a single parametrized test. Though I'm also not sure if it makes sense to have a separate test for redirected items now.